src/Card/CardHeader.js

```diff
@@ -37,32 +37,25 @@ export default function CardHeader(props, context) {
   const classes = context.styleManager.render(styleSheet);
   const className = classNames(classes.cardHeader, classNameProp);
 
-  if (avatar) {
-    return (
-      <CardContent className={className} {...other}>
-        <div className={classes.avatar}>
-          {avatar}
-        </div>
-        <div className={classes.content}>
-          <Text type="body2" gutterBottom>
-            {title}
-          </Text>
-          <Text type="body2" secondary className={classes.contentSecondary}>
-            {subheader}
-          </Text>
-        </div>
-      </CardContent>
-    );
-  }
+  // adjustments that depend on the presence of an avatar
+  const titleType = avatar ? 'body2' : 'headline';
+  const subheaderType = avatar ? 'body2' : 'body1';
 
   return (
     <CardContent className={className} {...other}>
-      <Text type="headline">
-        {title}
-      </Text>
-      <Text type="body1" secondary>
-        {subheader}
-      </Text>
+      {avatar &&
+        <div className={classes.avatar}>
+          {avatar}
+        </div>
+      }
+      <div className={classes.content}>
+        <Text type={titleType}>
```

Comment:
> The logic would be: if it's wrong, it's most likely that we are not following the leading typography specification for that size.

It's trickier than that. We are currently using the [leading values](https://material.io/guidelines/style/typography.html#typography-line-height) as CSS `line-height`. Leading (so called because historically strips of lead were placed between rows of manually set type)  is the distance between the baseline of subsequent lines of text, so `line-height` only achieves this correctly for multiple lines of the same font size.

`gutterBottom` (which we need to rename regardless) would need to calculate the bottom margin according to the height of the following row of text which, AFAIK, isn't possible.